### PR TITLE
5264 fix ghost tables deletion

### PR DESCRIPTION
**Pull request by rafatower**
A bit of explanation about this fix:
- Table.create_table_in_database! is a private method
- It is invoked from Table.before_create when it is not a migration nor
  a register of the table, that is, only when creating empty tables.
  E.g: from TablesController.create
- cartodbfy will be called a bit later, but it will be idempotent (no
  further changes).
- If you look into Table.before_create, it will invoke
  Table.set_table_id which is the right place to edit such a thing (the
  OID of the just created use rtable).

So this is not the perfect solution, but is the one that solves the
problem at hand minimizing side effects.

@Kartones or @juanignaciosl can you please review?

### CODE REVIEW

**Comment by Kartones** [comment]
:+1:  Looks fine, but I trust your judgement as is not easy to follow the whole import flow in all scenarios

File: app/models/table.rb [hunk 1]
```diff
@@ -1602,6 +1607,7 @@ def create_table_in_database!
         )
       end
     end
+    self.cartodbfy
```
Comment (juanignaciosl) [inline app/models/table.rb]: This is difficult to trace and you probably know the answer: why is there a _new_ cartodbfication without removing it elsewhere?

**Comment by juanignaciosl** [comment]
Added a question, but, as @Kartones said, not much can be said here :-\ :+1:

File: app/models/table.rb (hunk 1 above)
Comment (rafatower) [inline app/models/table.rb]: that new cartodbfy is right after actual table creation (no registration, no import...)

it is meant to be removed from after_create, but I do not dare to remove it yet in this changeset

**Comment by rafatower** [comment]
I'm reworking all this in another branch, closing this PR

**Pull request closed without merging**